### PR TITLE
feat(VDatePicker): `show-adjacent-months` prop

### LINKED ISSUE #12237: [Feature Request] v-date-picker: Possibility to show dates for previous and next month
### Problem to solve
I use the v-date-picker to select a week at a time. For weeks where the whole week is within the month, it works well, but for the time period 28 september - 4 october for instance, only 28, 29 and 30 will be visible.

### Proposed solution
I would like dates for previous month and next month which will fit within the table, to be visible: For september 2020: 31:st of August and 1-4:th of October would also show.

Compare with Ubuntu calendar for instance, which has this feature.

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Description
Adds possibility to display days from prev/next monhts

## Motivation and Context
fixes #12237

## How Has This Been Tested?
playground

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-date-picker
      v-model="dates"
      :picker-date.sync="pickerDate"
      multiple
      show-current="2021-01-02"
    />
    <v-date-picker
      v-model="dates"
      show-adjacent-months
      :picker-date.sync="pickerDate"
      multiple
      show-current="2021-01-02"
    />
  </v-container>
</template>

<script>
  export default {
    data: () => ({ dates: ['2020-11-29', '2021-01-01'], pickerDate: '2020-12' }),
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by nekosaur — DISMISSED** [review]
Looks good. The only thing I could see is that the text/background contrast ratio for the disabled selected dates is pretty bad, but I guess that's a separate issue.

**Comment by jacekkarczmarczyk** [comment]
Yeah, i'll try to fix in a separate PR

**Comment by KaelWD** [comment]
MD2 is kinda weird about this, the mobile picker doesn't show them:
image
The desktop picker does, but only the next month?
image

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts [hunk 1]
```diff
@@ -124,12 +124,12 @@ export default mixins(
...
       return this.$createElement('button', setColor(color, {
         staticClass: 'v-btn',
-        class: this.genButtonClasses(isAllowed, isFloating, isSelected, isCurrent),
+        class: this.genButtonClasses(isAllowed && !isOtherMonth, isFloating, isSelected, isCurrent),
         attrs: {
           type: 'button',
         },
         domProps: {
-          disabled: this.disabled || !isAllowed,
+          disabled: this.disabled || !isAllowed || isOtherMonth,
```
Comment (KaelWD) [inline packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts:132]: idk if disabled is right, the native ones are greyed out but still clickable: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

File: packages/vuetify/src/components/VDatePicker/VDatePicker.ts [hunk 2]
```diff
@@ -121,6 +121,7 @@ export default mixins(
       type: String,
       default: '$vuetify.datePicker.itemsSelected',
     },
+    showSiblingMonths: Boolean,
```
Comment (KaelWD) [inline packages/vuetify/src/components/VDatePicker/VDatePicker.ts:124]: `showAdjacentMonths`? "sibling" implies possibly not directly next to.

Comment (KaelWD) [reply packages/vuetify/src/components/VDatePicker/VDatePicker.ts:124]: Or just have it be default behaviour.

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VDatePicker/VDatePicker.ts:124]: Same question as for hold-on-hover for snackbar - do we need a prop to turn this behaviour off?

Comment (KaelWD) [reply packages/vuetify/src/components/VDatePicker/VDatePicker.ts:124]: I don't think so.

**Comment by jacekkarczmarczyk** [comment]
Renamed prop to `show-adjacent-months`

**Review by KaelWD — APPROVED** [review]

**Comment by graciaanna17** [comment]
Is it possible to enable the days reflected by shownAdjacentMonths?

**Comment by arcofine** [comment]
Can we add a prop to allow the shownAdjacentsMonths clickable if isAllowed?

**Pull request merged**